lib/DDG/Meta/Data.pm

```diff
@@ -25,25 +26,44 @@ unless(%ia_metadata){
 
     my @ia_types = qw(Spice Goodie Longtail Fathead);
 
-    # Load IA metadata files. Not all types are required during development.
-    my %metadata_files;
-    for my $iat (@ia_types){
-        my $bundle = "DDG::${iat}Bundle::OpenSourceDuckDuckGo";
-        eval "require $bundle";
-        my $f = eval{ dist_file("DDG-${iat}Bundle-OpenSourceDuckDuckGo", lc $iat . '/meta/metadata.json') }
-            or (debug && warn $@);
-        $metadata_files{$iat} = $f if $f;
+    my $tmpdir = io->tmpdir;
+    unless($tmpdir){
+       die 'No system temp directory found';
     }
 
-    unless(%metadata_files){
-        warn("[Error] No Instant Answer bundles installed. If you are developing an Instant Answer, please\n",
-             "install one or more of the following (via `duckpan` or `cpanm --mirror http://duckpan.org`),\n",
-         "including the type with which you are working:\n\n\t",
-        join("\n\t", map{ "DDG::${_}Bundle::OpenSourceDuckDuckGo" } @ia_types), "\n"); # and exit 1;
+    my $mdir = "$tmpdir/ddg-$>";
+    unless(-d $mdir){
+        mkdir $mdir or die "Failed to mkdir $mdir: $!";
     }
 
-    FILE: while (my ($type, $filename) = each %metadata_files) {
-        warn "Processing IA type: $type with file $filename" if debug;
+    # Load IA metadata. Not all types are required during development.
+    for my $iat (@ia_types){
+        debug && warn "Processing IA type: $iat";
+
+        my $json_endpt = lc $iat;
+        $json_endpt =~ s/goodie/goodies/;
+
+        # Prefer freshly downloaded metadata and fall back to metadata
+        # bundled with installed IA repos
+        my $f = "$mdir/$iat.json";
+        my $c = $ENV{NO_METADATA_DOWNLOAD} ? 0 : mirror("https://duck.co/ia/repo/$json_endpt/json", $f);
+        my $json;
+        if($c == RC_OK || $c == RC_NOT_MODIFIED){
+            $json = io($f)->slurp; 
+        }
+        else {
+            debug && warn "Failed to download metdata for $iat: $c";
+            if(-f $f){
+                $json = io($f)->all;
+            }
+            else{
+               warn "Failed to download metadata for $iat and no local file $f";
```

Comment:
I think we want to depend on /tmp as the fallback.  Whether it's a fatal error or not is up for debate, e.g. when only running spices do we really need the Longtail json?